### ISSUE TITLE
Printing via ESC/POS fails with TypeError on PosBox

### ISSUE THREAD

**Issue by tobru** [odoo/odoo#25607]
Impacted versions:

Odoo 11.0-20180122 with Docker (official image from Docker Hub)

Steps to reproduce:

Run Odoo with Docker and install the [hw_printer_network](https://github.com/it-projects-llc/pos-addons/tree/11.0/hw_printer_network) and [pos_printer_network](https://github.com/it-projects-llc/pos-addons/tree/11.0/pos_printer_network) add-ons. Configure PosBox and a network PoS printer.

Current behavior:

When printing a receipt, it fails:

```
posbox_1  | 2018-07-04 19:40:58,896 1 INFO ? werkzeug: 172.22.0.1 - - [04/Jul/2018 19:40:58] "POST /hw_proxy/print_xml_receipt?session_id= HTTP/1.1" 200 -
posbox_1  | 2018-07-04 19:40:58,933 1 INFO ? odoo.addons.hw_printer_network.controllers.hw_printer_network_controller: Printing XML receipt on EPSON printer 192.168.25.118...
posbox_1  | 2018-07-04 19:40:58,976 1 INFO ? odoo.addons.hw_escpos.controllers.main: error : Can't convert 'bytes' object to str implicitly
posbox_1  | 2018-07-04 19:40:58,978 1 ERROR ? odoo.addons.hw_escpos.controllers.main: ESC/POS Error: Can't convert 'bytes' object to str implicitly
posbox_1  | 2018-07-04 19:40:58,981 1 ERROR ? odoo.addons.hw_printer_network.controllers.hw_printer_network_controller: Can't convert 'bytes' object to str implicitly
posbox_1  | ------------------------------------------------------------
posbox_1  | Traceback (most recent call last):
posbox_1  |   File "/usr/lib/python3/dist-packages/odoo/addons/hw_escpos/escpos/escpos.py", line 698, in receipt
posbox_1  |     self._raw(stylestack.to_escpos())
posbox_1  |   File "/usr/lib/python3/dist-packages/odoo/addons/hw_escpos/escpos/printer.py", line 214, in _raw
posbox_1  |     self.device.send(msg)
posbox_1  | TypeError: a bytes-like object is required, not 'str'
posbox_1  | 
posbox_1  | During handling of the above exception, another exception occurred:
posbox_1  | 
posbox_1  | Traceback (most recent call last):
posbox_1  |   File "/mnt/extra-addons/hw_printer_network/controllers/hw_printer_network_controller.py", line 119, in run
posbox_1  |     printer.receipt(receipt)
posbox_1  |   File "/usr/lib/python3/dist-packages/odoo/addons/hw_escpos/escpos/escpos.py", line 710, in receipt
posbox_1  |     self.text(errmsg)
posbox_1  |   File "/usr/lib/python3/dist-packages/odoo/addons/hw_escpos/escpos/escpos.py", line 813, in text
posbox_1  |     txt = encode_str(txt)
posbox_1  |   File "/usr/lib/python3/dist-packages/odoo/addons/hw_escpos/escpos/escpos.py", line 810, in encode_str
posbox_1  |     buffer += encode_char(c)
posbox_1  |   File "/usr/lib/python3/dist-packages/odoo/addons/hw_escpos/escpos/escpos.py", line 803, in encode_char
posbox_1  |     encoded = encodings[encoding] + encoded
posbox_1  | TypeError: Can't convert 'bytes' object to str implicitly
posbox_1  | ------------------------------------------------------------
posbox_1  | 
```

Expected behavior:

No exceptions and receipt is printed

Video/Screenshot link (optional):

**Comment by pedrobaeza** [comment]
AFAIK, PosBox can't run on latest Odoo v11. @mart-e can you confirm this?

**Comment by mart-e** [comment]
Indeed, so far the posbox runs on a version 8 of Odoo, it is not converted to python3.
It is not an issue to have a different version on the server and posbox.
We do plan to migrate it, there is a branch at #16682 to migrate it to v10 but not yet to v11.

**Comment by pedrobaeza** [comment]
Then that's the problem @tobru. Closing as not supported yet. Please check the PR indicated for knowing the progress.

**Comment by tobru** [comment]
I can confirm that with Odoo 8 it just works :tm: It's a bit misleading that Odoo 8 is not supported anymore (also on Docker Hub not available anymore).

**Comment by tobru** [comment]
Thanks @mart-e for the heads-up. I got it running with Odoo 8 sources and it works as expected.

**Comment by mart-e** [comment]
FYI, we have work to move it to python3/v11 at #25711

**Comment by tobru** [comment]
That's great, looking forward to it. Thank you very much...

**Comment by mart-e** [comment]
Good news, we have merged #25711 
@tobru It would help us a lot if we could get some people to test our new image?

We have published the latest image (v17_beta) on https://nightly.odoo.com/trunk/posbox/
If you are using a posbox-less version, just updating the source code should do it.

Thanks !

**Comment by amendoncabh** [comment]
I just make adjust in printer.py _raw funciton...

    def _raw(self, msg):
        if type(msg) == str:
            msg = bytes(msg, 'UTF-8')
        self.device.send(msg)
